### PR TITLE
Add surrogate pair support, fixes #69

### LINKED ISSUE #69: Does not handle surrogate pairs correctly
[For example](https://runkit.com/embed/r2tg5gjo7vzo), it will output `"var𫠝"` for the AST given by `parseScript('var 𫠝')`.

shift-java had the same problem; [this PR](https://github.com/shapesecurity/shift-java/pull/178/files) fixed it.

The problem is that the "check if the last thing was IdentifierStart and this thing is IdentifierContinue and, if so, output a space" logic is not mindful of surrogate pairs, so its notion of "this thing" can be wrong.

### PR DESCRIPTION
**Pull request by Protryon**

### CODE REVIEW

File: src/token-stream.js [hunk 1]
```diff
@@ -16,6 +16,21 @@
...
+
+function isIdentifierPartES6(char) {
+  let charCode = char.codePointAt(0);
+  if (charCode < 0xFFFF && code.isIdentifierPartES6(charCode)) {
+    return true;
+  }
+  if (char.match(IDENTIFIER_CONTINUE) != null) {
```
Comment (michaelficarra) [inline src/token-stream.js]:
```js
return IDENTIFIER_CONTINUE.test(char)
```

File: src/token-stream.js [hunk 2]
```diff
@@ -16,6 +16,21 @@
 
 import { code } from 'esutils';
 
+// following constant is generated by shift-java/scripts/make-unicode-regexen.js, IDENTIFIER_CONTINUE
```
Comment (michaelficarra) [inline src/token-stream.js]: Probably refer to this one instead? https://github.com/shapesecurity/shift-parser-js/blob/es2016/scripts/generate-unicode-data.js

Comment (michaelficarra) [reply src/token-stream.js]: Also let's just actually save the whole generated file, keep it marked as generated, and import it here.

File: src/token-stream.js (hunk 1 above)
Comment (michaelficarra) [reply src/token-stream.js]: Actually, what does this even do beyond `code.isIdentifierPartES6`? I think that's sufficient.

Comment (Protryon) [reply src/token-stream.js]: @michaelficarra `code.isIdentifierPartES6` doesn't support surrogate pairs.

File: src/unicode.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// Generated by scripts/generate-unicode-data.js
```
Comment (michaelficarra) [inline src/unicode.js:1]: Technically `shapesecurity/shift-parser-js/scripts/generate-unicode-data.js`.

File: .gitattributes [hunk 4]
```diff
@@ -0,0 +1 @@
+src/token-stream.js linguist-generated=false
```
Comment (michaelficarra) [inline .gitattributes]: You don't need this anymore.

**Review by michaelficarra — APPROVED** [review]

**Pull request merged**